File: backend/voting-app/src/main/java/com/tu/votingapp/utils/mappers/LocationMapper.java
package com.tu.votingapp.utils.mappers;

import com.tu.votingapp.dto.general.LocationDTO;
import com.tu.votingapp.entities.LocationEntity;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring", uses = {MunicipalityMapper.class})
public interface LocationMapper {
    LocationDTO toDto(LocationEntity entity);
    LocationEntity toEntity(LocationDTO dto);
}

File: frontend/android/app/src/main/java/com/example/android/utils/mappers/LocationMapper.kt
package com.example.android.utils.mappers

import com.example.android.dto.response.LocationResponseDTO
import com.example.android.entity.LocationEntity
import com.example.android.entity.MunicipalityEntity
import com.example.android.entity.RegionEntity
import com.example.android.utils.mappers.MunicipalityMapper.toDto

object LocationMapper {
    fun LocationResponseDTO.toEntity(): LocationEntity = LocationEntity(
        id = this.id,
        name = this.name,
        municipalityId = municipality.id
    )

    fun LocationEntity.toDto(municipality: MunicipalityEntity, region: RegionEntity): LocationResponseDTO = LocationResponseDTO(
        id = this.id!!,
        name = this.name,
        municipality = municipality.toDto(region)
    )
}